core/ledger/kvledger/channelinfo_provider.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package kvledger
+
+import (
+	cb "github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/bccsp/factory"
+	"github.com/hyperledger/fabric/common/channelconfig"
+	"github.com/hyperledger/fabric/common/ledger/blkstorage"
+	"github.com/hyperledger/fabric/protoutil"
+	"github.com/pkg/errors"
+)
+
+type channelInfoProvider struct {
+	channelName string
+	blockStore  *blkstorage.BlockStore
+}
+
+// GetAllMSPIDs retrieves the MSPIDs of application organizations in all the channel configurations,
+// including current and previous channel configurations.
+func (p *channelInfoProvider) GetAllMSPIDs() ([]string, error) {
+	blockchainInfo, err := p.blockStore.GetBlockchainInfo()
+	if err != nil {
+		return nil, err
+	}
+	if blockchainInfo.Height == 0 {
+		return nil, nil
+	}
+
+	// Iterate over the config blocks to get all the channel configs, extract MSPIDs and add to mspidsMap
+	mspidsMap := map[string]string{}
+	blockNum := blockchainInfo.Height - 1
+	for {
+		configBlock, err := mostRecentConfigBlockAsOf(blockNum, p.blockStore)
+		if err != nil {
+			return nil, err
+		}
+		channelConfig, err := channelconfig.ExtractChannelConfig(configBlock, factory.GetDefault())
+		if err != nil {
+			return nil, err
+		}
+
+		if channelConfig.ApplicationConfig() == nil {
+			return nil, errors.Errorf("could not get application config for channel '%s'", p.channelName)
+		}
+		mspids := channelConfig.ApplicationConfig().MSPIDs()
+		for _, mspid := range mspids {
+			if _, ok := mspidsMap[mspid]; !ok {
+				mspidsMap[mspid] = ""
+			}
+		}
+
+		if configBlock.Header.Number == 0 {
+			break
+		}
+		blockNum = configBlock.Header.Number - 1
```

Comment:
Just to confirm - A config block (other than the genesis block) refers to itself as the `last config block` and not the previous one?

Reply:
Yes, a config block refers to itself as the last config block, not the previous one.